### PR TITLE
fix: strip cookie header on redirect across hostnames

### PR DESCRIPTION
**Pull request by wraithgar**
Fixes https://github.com/npm/make-fetch-happen/issues/70

### CODE REVIEW

File: test/fetch.js [hunk 1]
```diff
@@ -317,6 +317,36 @@ t.test('removes authorization header if changing hostnames', async (t) => {
...
+    .reply(301, '', { Location: `${HOST}/test` })
+
+  const httpsSrv = nock(HOST, {
+    reqheaders: {
+      cookie: 'test=true',
+    },
+  })
+    .get('/test')
+    .reply(200, () => {
+      t.equal(true, false, 'meaningful failure, this should never be executed')
```
Comment (ruyadorno) [inline test/fetch.js:336]: can also use `t.fail()` here instead 😊

**Review by ruyadorno — APPROVED** [review]
👍 LGTM

**Review by nlf — APPROVED** [review]

**Comment by ruyadorno** [comment]
it looks like node10 ci is failing due to usage of private fields? I assume node10 is no longer supported by this package?

we should def update the package to note the correct engines requirement and remove the node10 ci target

**Comment by wraithgar** [comment]
@ruyadorno getting this off node10 is on the map for today but I wanted to land extant PRs first

**Comment by ruyadorno** [comment]
absolutely! just mentioning as a note for the future! 😄 good to know it's already in the radar

File: test/fetch.js (hunk 1 above)
Comment (wraithgar) [reply test/fetch.js:336]: I copied the other header test, this was not the PR in which to be iterating on the tests

**Pull request merged**